### DISCUSSION TITLE
Configuring Workflows and State Transitions in GlobaLeaks workflow

### DISCUSSION THREAD

**Discussion by Ridon77** [globaleaks/globaleaks-whistleblowing-software#4130] [Q&A]
Hello everyone,

I am currently working on configuring workflows in GlobaLeaks and need assistance with defining transitions between the states of reports. I would like to know how I can set restrictions to control which states are accessible from a particular state.

What I Have Achieved So Far:
State Definitions:

I have created the following states for reports:
New
Under Review
Pending Information
Closed
Rejected
Basic Transitions:

What I Need Help With:
Transition Restrictions:

I want to ensure that a report in the "Pending Information" state can only move to "Under Review" if the required additional information has been provided.
How can I configure these conditions in the Globaleaks interface?
Best Practices:

Are there any recommendations or best practices for configuring workflows and transitions in GlobalLeaks to ensure efficient and secure handling of reports?
Example of Desired Configuration:
From "New":

Allow transition to "Under Review"
Allow transition to "Rejected"
From "Under Review":

Allow transition to "Pending Information"
Allow transition to "Closed"
Allow transition to "Rejected"
From "Pending Information":

Allow transition to "Under Review" only if the additional information is complete.
From "Closed" and "Rejected":

Do not allow transition to other states (end of the workflow).
I appreciate any advice, documentation, or experiences you can share about how to configure these transitions and conditions in GlobalLeaks. Thanks in advance for your help!

Best regards,
Joan Redon

**Comment by evilaliv3** [comment]
Thank you @Ridon77 

At the moment such a feauture of transition does do not exist.

Would you like to detail it further and support its implementation ?

**Comment by evilaliv3** [comment]
p.s.: GlobaLeaks is written with just one L. Please pass it on.

**Comment by Ridon77** [comment]
Thanks,
Will this feature be developed in the future?

How can I help in the implementation?

My Idea is at least define for each state wich states are available, Can be nice to be able to define restrictions (mandatory values to be informed) in case of some transition. 

For example, define from where a case can be "closed" and only if "resolution info" to be informed. 

We can talk if you need more info or want to talk about it?

Sorry for mistake on Globaleaks name.

**Comment by evilaliv3** [comment]
Thank you @Ridon77 

Yes the feature could be implemented.

GlobaLeaks is a community based project where we keep in consideration every user feedback.
You could support its implementation with some mockups, direct development or eventually with some donations or complete funding for the features you need  and we may try to find developers that could support this effort.

**Reply by Ridon77** [reply]
Sorry for not getting back to you sooner.

I can help with an analysis of how it can work easily. 

Do you prefer to send in this thread or other way?
